### PR TITLE
Create lightweight resource provider for managing a YAJSW application

### PR DESCRIPTION
**Pull request by fishnix**
This adds a lightweight resource provider for managing a YAJSW application.  This will allow us to use this code more modularly (and not set node-wide attributes in the node['yajsw']['xxxx'] namespace.  Instead, I can reference the LWRP from the platform cookbook:

```
yajsw_app 'myapp' do
  some_option node['np-platform']['myapp']['xxxx']
end
```

### CODE REVIEW

**Comment by cgatesman** [comment]
Looks good! :+1:

File: providers/app.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      :mainclass => new_resource.mainclass,
+      :initmemory => new_resource.initmemory,
+      :maxmemory => new_resource.maxmemory,
+      :logfile => new_resource.logfile
+    })
+    owner appuser
+    mode 0755
+  end
+  new_resource.updated_by_last_action(t.updated_by_last_action?)
+end
```
Comment (jwinfield) [inline providers/app.rb]: It might be a good idea to save a backup of an existing wrapper.conf in case it was hand edited.

**Comment by jwinfield** [comment]
That's all I had. :+1:

File: providers/app.rb (hunk 1 above)
Comment (fishnix) [inline providers/app.rb]: hand edited!  :scream: sure - i can tell it to keep a couple backups ;)

**Pull request merged**